soroban-sdk/src/contract_event.rs

```diff
@@ -0,0 +1,66 @@
+use core::fmt::Debug;
+
+use crate::{env::internal, Env, IntoVal, RawVal, Vec};
+
+/// ### Examples
+///
+/// ```
+/// use soroban_sdk::Env;
+///
+/// # use soroban_sdk::{contractimpl, vec, map, RawVal, FixedBinary};
+/// #
+/// # pub struct Contract;
+/// #
+/// # #[contractimpl]
+/// # impl Contract {
+/// #     pub fn f(env: Env) {
+/// let event = env.contract_event();
+/// let topics = vec![&env, 0u32, 1u32];
+/// let data = map![&env, (1u32, 2u32)];
+/// event.publish(topics, data)
+/// #     }
+/// # }
+///
+/// # #[cfg(feature = "testutils")]
+/// # fn main() {
+/// #     let env = Env::default();
+/// #     let contract_id = FixedBinary::from_array(&env, [0; 32]);
+/// #     env.register_contract(&contract_id, Contract);
+/// #     f::invoke(&env, &contract_id);
+/// # }
+/// # #[cfg(not(feature = "testutils"))]
+/// # fn main() { }
+/// ```
+#[derive(Clone)]
+pub struct ContractEvent(Env);
```

Comment:
I think it might be worth naming this `ContractEvents`, or maybe even just `Events`, since it doesn't represent a single event and instead represents contract events the functionality. I don't feel strongly.